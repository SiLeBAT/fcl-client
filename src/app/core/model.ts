export interface ValidationError {
    code: string;
    message: string;
}
