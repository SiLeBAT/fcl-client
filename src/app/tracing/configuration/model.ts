import {
    Color,
    DataServiceInputState,
    LabelPart,
    NodeShapeType,
    TableColumn,
    TableRow,
} from "../data.model";
import { ComplexFilterCondition, PropToValuesMap } from "./configuration.model";

export interface RowFilter<T> {
    filter(arr: TableRow[]): TableRow[];
    getSettings(): T;
}

export interface TextFilterSettings {
    filterTerm: string;
    filterProps: string[];
}

export enum TableType {
    STATIONS,
    DELIVERIES,
}

export type TreeStatus = "collapsed" | "expanded";

export interface EditHighlightingServiceData {
    favouriteProperties: TableColumn[];
    otherProperties: TableColumn[];
    propToValuesMap: PropToValuesMap;
    ruleListItems: RuleListItem[];
}

export interface RuleListItem {
    id: RuleId;
    name: string;
    color: Color | null;
    shape: NodeShapeType | null;
    showInLegend: boolean;
    disabled: boolean;
    autoDisabled: boolean;
    isAnonymizationRule: boolean;
    effElementsCountTooltip: string;
    effElementsCount: number;
    conflictCount: number;
    ruleType: RuleType | null;
}

export interface EditRuleCore {
    id: string;
    name: string;
    userDisabled: boolean;
    complexFilterConditions: ComplexFilterCondition[];
    isValid: boolean;
}

export type EditRule =
    | ColorAndShapeEditRule
    | ColorEditRule
    | LabelEditRule
    | InvEditRule;

export enum RuleType {
    INVISIBILITY,
    LABEL,
    COLOR_AND_SHAPE,
    COLOR,
}

export interface ColorAndShapeEditRule extends EditRuleCore {
    type: RuleType.COLOR_AND_SHAPE;
    showInLegend: boolean;
    color: Color | null;
    shape: NodeShapeType | null;
}

export interface ColorEditRule extends EditRuleCore {
    type: RuleType.COLOR;
    showInLegend: boolean;
    color: Color;
}

export interface SimpleLabelEditRule extends EditRuleCore {
    type: RuleType.LABEL;
    labelProperty: string | null;
}

export interface ComposedLabelEditRule extends EditRuleCore {
    type: RuleType.LABEL;
    labelParts: LabelPart[];
    labelPrefix: string;
}

export type LabelEditRule = SimpleLabelEditRule | ComposedLabelEditRule;

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface InvEditRule extends EditRuleCore {
    type: RuleType.INVISIBILITY;
}

export type StationEditRule =
    | ColorAndShapeEditRule
    | LabelEditRule
    | InvEditRule;
export type DeliveryEditRule = LabelEditRule | InvEditRule | ColorEditRule;

export type DeliveryRuleType = RuleType.LABEL | RuleType.COLOR;
export type StationRuleType = RuleType.LABEL | RuleType.COLOR_AND_SHAPE;

export type RuleId = string;

export interface EditHighlightingState<T extends EditRule> {
    dataServiceInputState: DataServiceInputState;
    editRules: T[];
}
