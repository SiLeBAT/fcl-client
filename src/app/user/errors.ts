import { InvalidServerInputHttpErrorResponse } from '@app/core/errors';
import { ValidationError } from '@app/core/model';
import { CODE_TO_FIELD_MAP } from './consts/error-code-mappings.consts';
import { RegistrationCredentials } from './models/user.model';

export class InvalidRegistrationInput {
    private _errors: Partial<Record<keyof RegistrationCredentials, ValidationError[]>> = {};
    private _message: string | null = null;

    get errors(): Partial<Record<keyof RegistrationCredentials, ValidationError[]>> {
        return this._errors;
    }

    get message(): string | null {
        return this._message;
    }

    constructor(error: InvalidServerInputHttpErrorResponse) {
        this._message = error.message;
        this.mapErrors(error.errors);
    }

    private mapErrors(validationErrors: ValidationError[]): void {
        this._errors = {};
        validationErrors.forEach(error => {
            const field = CODE_TO_FIELD_MAP[error.code] as undefined | keyof RegistrationCredentials;
            if (field !== undefined) {
                const fieldErrors = this._errors[field];
                if (fieldErrors !== undefined) {
                    fieldErrors.push(error);
                } else {
                    this._errors[field] = [error];
                }
            }
        });
    }
}
