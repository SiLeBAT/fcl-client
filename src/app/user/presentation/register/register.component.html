<ng-template let-fieldName="fieldName" #serverValidationErrorsTemplate>
    <ng-container *ngFor="let error of serverValidationErrors[fieldName]; let i=index">
        <br *ngIf="i>0"><span>{{ error.message }}</span>
    </ng-container>
</ng-template>

<form [formGroup]="registerForm" class="fcl-register-form" (ngSubmit)="onRegister()">
    <div class="fcl-mat-card-content">
        <div class="form-group fcl-form-firstname">
            <mat-form-field class="fcl-form-field" appearance="standard">
                <mat-label>First Name</mat-label>
                <input matInput formControlName="firstName" name="firstName" required/>
                <mat-icon matPrefix>account_circle</mat-icon>
                <mat-error *ngIf="!validateField('firstName') && registerForm.controls.firstName.errors.required && !serverValidationErrors.firstName">
                    Required Field
                </mat-error>
                <mat-hint *ngIf="serverValidationErrors.firstName" class="mat-error">
                    <ng-container [ngTemplateOutlet]="serverValidationErrorsTemplate"
                        [ngTemplateOutletContext] ="{ fieldName: 'firstName' }">
                    </ng-container>
                </mat-hint>
            </mat-form-field>
        </div>

        <div class="form-group">
            <mat-form-field class="fcl-form-field" appearance="standard">
                <mat-label>Last Name</mat-label>
                <input matInput formControlName="lastName" name="lastName" required/>
                <mat-icon matPrefix>account_circle</mat-icon>
                <mat-error *ngIf="!validateField('lastName') && registerForm.controls.lastName.errors.required && !serverValidationErrors.lastName">
                    Required Field
                </mat-error>
                <mat-hint *ngIf="serverValidationErrors.lastName" class="mat-error">
                    <ng-container [ngTemplateOutlet]="serverValidationErrorsTemplate"
                        [ngTemplateOutletContext] ="{ fieldName: 'lastName' }">
                    </ng-container>
                </mat-hint>
            </mat-form-field>
        </div>

        <div class="form-group">
            <mat-form-field class="fcl-form-field" appearance="standard">
                <mat-label>Email</mat-label>
                <input matInput formControlName="email" name="email" required/>
                <mat-icon matPrefix>mail</mat-icon>
                <mat-error *ngIf="!validateField('email') && registerForm.controls.email.errors.required">
                    Required field
                </mat-error>
                <mat-error *ngIf="!validateField('email') && registerForm.controls.email.errors.email">
                    Not a valid email
                </mat-error>
                <mat-hint *ngIf="serverValidationErrors.email" class="mat-error">
                   <ng-container [ngTemplateOutlet]="serverValidationErrorsTemplate"
                        [ngTemplateOutletContext] ="{ fieldName: 'email' }">
                    </ng-container>
                </mat-hint>
            </mat-form-field>
        </div>

        <div class="form-group fcl-form-password" [ngClass]="{ error: !validateField('password1') }">
            <mat-form-field class="fcl-form-field" appearance="standard">
                <mat-label>Password</mat-label>
                <input matInput #message formControlName="password1" type="password" name="password1" required/>
                <mat-icon matPrefix>lock</mat-icon>
                <mat-error *ngIf="!validateField('password1') && registerForm.controls.password1.errors.required && !serverValidationErrors.password">
                    Required field
                </mat-error>
                <mat-hint *ngIf="serverValidationErrors.password" class="mat-error">
                    <ng-container [ngTemplateOutlet]="serverValidationErrorsTemplate"
                        [ngTemplateOutletContext] ="{ fieldName: 'password' }">
                    </ng-container>
                </mat-hint>
            </mat-form-field>
        </div>
        <div class="form-group">
            <mat-form-field class="fcl-form-field" appearance="standard">
                <mat-label>Confirm Password</mat-label>
                <input matInput formControlName="password2" type="password" name="password2" required />
                <mat-icon matPrefix>lock</mat-icon>
                <mat-error *ngIf="!validateField('password2') && registerForm.controls.password2.errors.validatePasswordConfirm">
                    Passwords must match
                </mat-error>
            </mat-form-field>
        </div>

        <div class="form-group fcl-form-checkbox">
            <mat-checkbox required formControlName="dataProtection">
                I have read and accept the
                <a
                    class="fcl-policy-link"
                    target="_blank"
                    [routerLink]="['/content/dataprotectiondeclaration']">
                    Privacy Policy
                </a>
                *
            </mat-checkbox>
            <mat-error *ngIf="!validateField('dataProtection')">
                You have to accept the Privacy Policy
            </mat-error>
        </div>

        <div class="form-group fcl-form-checkbox">
            <mat-checkbox
                class="fcl-newsletter-checkbox"
                formControlName="newsletter">
                <div class="fcl-text-wrap">
                    I agree to receive occasional email newsletters on e.g. new features of FCL Web or user trainings.
                    I can unsubscribe anytime by sending an email to
                    <a
                        class="fcl-policy-link"
                        href="mailto:foodrisklabs@bfr.bund.de" target="_top">
                        foodrisklabs@bfr.bund.de
                    </a>
                    .
                </div>
            </mat-checkbox>
        </div>

        <div class="fcl-card-action fcl-card-action-container">
            <div class="form-group fcl-submit-button">
                <button mat-raised-button color="primary" type="submit" [disabled]="!registerForm.valid">
                    Submit
                </button>
            </div>
            <div>
                <a mat-button [routerLink]="['/users/login']">
                    Cancel
                </a>
            </div>
        </div>
    </div>
</form>
