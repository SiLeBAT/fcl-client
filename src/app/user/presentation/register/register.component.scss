@use "../../../../assets/sass/base" as fcl;

.fcl-card-action-container {
    display: flex;
}

.fcl-mat-card-content {
    width: 500px;
}

.fcl-form-firstname {
    margin-top: 2rem;
}

.fcl-form-password,
.fcl-form-checkbox {
    margin-top: 1.5rem;
}

.fcl-policy-link {
    text-decoration: none;
}

.fcl-card-action {
    margin-top: 3rem;
}

.fcl-submit-button {
    margin-right: fcl.$fcl-toolbar-padding * 4;
}

.fcl-form-field {
    width: 100%;
}

.fcl-text-wrap {
    white-space: normal;
}

.fcl-newsletter-checkbox.mat-accent .mat-checkbox-inner-container {
    margin: 0.25rem 0.5rem auto 0;
}

em {
    @include fcl.em-errors;
}
