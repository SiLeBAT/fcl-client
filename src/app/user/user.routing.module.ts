import { NgModule } from "@angular/core";
import { Routes, RouterModule, mapToCanActivate } from "@angular/router";

import { LoginViewComponent } from "./presentation/login-view/login-view.component";
import { RegisterViewComponent } from "./presentation/register-view/register-view.component";
import { RecoveryViewComponent } from "./presentation/recovery-view/recovery-view.component";
import { ResetViewComponent } from "./presentation/reset-view/reset-view.component";
import { ActivateViewComponent } from "./presentation/activate-view/activate-view.component";
import { AdminActivateViewComponent } from "./presentation/admin-activate-view/admin-activate-view.component";
import { ProfileContainerComponent } from "./container/profile-container/profile-container.component";
import { environment } from "./../../environments/environment";
import { AuthGuard } from "./guards/auth.guard";
import { NewsActivateViewComponent } from "./presentation/news-activate-view/news-activate-view.component";

const userRoutes: Routes = [
    {
        path: "users",
        children: [
            {
                path: "profile",
                component: ProfileContainerComponent,
                canActivate: environment.serverless
                    ? undefined
                    : mapToCanActivate([AuthGuard]),
            },
            {
                path: "login",
                component: LoginViewComponent,
            },
            {
                path: "register",
                component: RegisterViewComponent,
            },
            {
                path: "recovery",
                component: RecoveryViewComponent,
            },
            {
                path: "reset/:id",
                component: ResetViewComponent,
            },
            {
                path: "activate/:id",
                component: ActivateViewComponent,
            },
            {
                path: "adminactivate/:id",
                component: AdminActivateViewComponent,
            },
            {
                path: "newsactivate/:id",
                component: NewsActivateViewComponent,
            },
        ],
    },
];

@NgModule({
    imports: [RouterModule.forChild(userRoutes)],
    exports: [RouterModule],
})
export class UserRoutingModule {}
