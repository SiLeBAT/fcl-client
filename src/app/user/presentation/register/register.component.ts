import {Component, Output, EventEmitter, Input} from '@angular/core';
import {
  Validators,
  ValidationErrors,
  FormControl,
  FormGroup,
  ValidatorFn,
} from '@angular/forms';
import {ValidationError} from '@app/core/model';
import {RegistrationCredentials} from '../../models/user.model';

interface RegisterForm {
  firstName: FormControl<string | null>;
  lastName: FormControl<string | null>;
  email: FormControl<string | null>;
  password1: FormControl<string | null>;
  password2: FormControl<string | null>;
  dataProtection: FormControl<boolean | null>;
  newsletter: FormControl<boolean | null>;
}

@Component({
  selector: 'fcl-register',
  templateUrl: './register.component.html',
  styleUrls: ['./register.component.scss'],
})
export class RegisterComponent {
  @Input() serverValidationErrors: Partial<
    Record<keyof RegistrationCredentials, ValidationError[]>
  > = {};

  @Output() register = new EventEmitter<RegistrationCredentials>();

  registerForm = new FormGroup<RegisterForm>(
    {
      firstName: new FormControl<string | null>(null, Validators.required),
      lastName: new FormControl<string | null>(null, Validators.required),
      email: new FormControl<string | null>(null, [
        Validators.required,
        Validators.email,
      ]),
      password1: new FormControl<string | null>(null, Validators.required),
      password2: new FormControl<string | null>(null),
      dataProtection: new FormControl<boolean | null>(
        null,
        Validators.requiredTrue
      ),
      newsletter: new FormControl<boolean | null>(false),
    },
    this.passwordConfirmationValidator as ValidatorFn
  );

  onRegister() {
    if (this.registerForm.valid) {
      const registrationCredentials: RegistrationCredentials = {
        email: this.registerForm.value.email ?? '',
        password: this.registerForm.value.password1 ?? '',
        firstName: this.registerForm.value.firstName ?? '',
        lastName: this.registerForm.value.lastName ?? '',
        dataProtectionAgreed: this.registerForm.value.dataProtection ?? false,
        newsRegAgreed: this.registerForm.value.newsletter ?? false,
        newsMailAgreed: false,
      };
      this.register.emit(registrationCredentials);
    }
  }

  validateField(fieldName: string): boolean {
    return (
      this.registerForm.controls[fieldName]?.valid ||
      this.registerForm.controls[fieldName]?.untouched
    );
  }

  private passwordConfirmationValidator(
    fg: FormGroup<RegisterForm>
  ): ValidationErrors | null {
    const pw1 = fg.controls.password1;
    const pw2 = fg.controls.password2;

    if (pw1.value !== pw2.value) {
      pw2.setErrors({validatePasswordConfirm: true});
    } else {
      pw2.setErrors(null);
    }
    return null;
  }
}
