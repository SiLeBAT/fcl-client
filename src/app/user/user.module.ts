import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { PasswordStrengthMeterModule } from 'angular-password-strength-meter';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { SharedModule } from './../shared/shared.module';
import { CoreModule } from '../core/core.module';
import { MainPageModule } from '../main-page/main-page.module';
import { UserRoutingModule } from './user.routing.module';
import { ContentModule } from '@app/content/content.module';

import { STATE_SLICE_NAME, reducer } from './state/user.reducer';
import { UserEffects } from './state/user.effects';
import { LoginContainerComponent } from './container/login-container/login-container.component';
import { LoginComponent } from './presentation/login/login.component';
import { LoginViewComponent } from './presentation/login-view/login-view.component';
import { RegisterContainerComponent } from './container/register-container/register-container.component';
import { RegisterComponent } from './presentation/register/register.component';
import { RegisterViewComponent } from './presentation/register-view/register-view.component';
import { RecoveryContainerComponent } from './container/recovery-container/recovery-container.component';
import { RecoveryComponent } from './presentation/recovery/recovery.component';
import { RecoveryViewComponent } from './presentation/recovery-view/recovery-view.component';
import { ResetContainerComponent } from './container/reset-container/reset-container.component';
import { ResetComponent } from './presentation/reset/reset.component';
import { ResetViewComponent } from './presentation/reset-view/reset-view.component';
import { ActivateContainerComponent } from './container/activate-container/activate-container.component';
import { ActivateComponent } from './presentation/activate/activate.component';
import { ActivateViewComponent } from './presentation/activate-view/activate-view.component';
import { AdminActivateContainerComponent } from './container/admin-activate-container/admin-activate-container.component';
import { AdminActivateComponent } from './presentation/admin-activate/admin-activate.component';
import { AdminActivateViewComponent } from './presentation/admin-activate-view/admin-activate-view.component';
import { ProfileContainerComponent } from './container/profile-container/profile-container.component';
import { ProfileComponent } from './presentation/profile/profile.component';
import { GdprAgreementComponent } from './presentation/gdpr-agreement/gdpr-agreement.component';
import { NewsActivateContainerComponent } from './container/news-activate-container/news-activate-container.component';
import { NewsActivateComponent } from './presentation/news-activate/news-activate.component';
import { NewsActivateViewComponent } from './presentation/news-activate-view/news-activate-view.component';

@NgModule({
    imports: [
        CommonModule,
        ContentModule,
        ReactiveFormsModule,
        PasswordStrengthMeterModule,
        SharedModule,
        CoreModule,
        MainPageModule,
        UserRoutingModule,
        StoreModule.forFeature(STATE_SLICE_NAME, reducer),
        EffectsModule.forFeature([UserEffects]),
        FormsModule
    ],
    exports: [],
    declarations: [
        LoginContainerComponent,
        LoginComponent,
        LoginViewComponent,
        RegisterContainerComponent,
        RegisterComponent,
        RegisterViewComponent,
        RecoveryViewComponent,
        RecoveryComponent,
        RecoveryContainerComponent,
        ResetContainerComponent,
        ResetComponent,
        ResetViewComponent,
        ActivateContainerComponent,
        ActivateComponent,
        ActivateViewComponent,
        AdminActivateContainerComponent,
        AdminActivateComponent,
        AdminActivateViewComponent,
        ProfileContainerComponent,
        ProfileComponent,
        GdprAgreementComponent,
        NewsActivateContainerComponent,
        NewsActivateComponent,
        NewsActivateViewComponent
    ],
    schemas: [CUSTOM_ELEMENTS_SCHEMA]

})
export class UserModule { }
