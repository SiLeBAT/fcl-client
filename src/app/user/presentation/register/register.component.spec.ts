import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { RegisterComponent } from './register.component';
import { MaterialModule } from '../../../shared/material.module';
import { SharedModule } from '../../../shared/shared.module';
import { RouterTestingModule } from '@angular/router/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RegistrationCredentials } from '../../models/user.model';

describe('RegisterComponent', () => {

    let component: RegisterComponent;
    let fixture: ComponentFixture<RegisterComponent>;

    beforeEach(async(() => {
        // eslint-disable-next-line @typescript-eslint/no-floating-promises
        TestBed.configureTestingModule({
            declarations: [
                RegisterComponent
            ],
            imports: [
                MaterialModule,
                SharedModule,
                RouterTestingModule,
                NoopAnimationsModule
            ]
        }).compileComponents()
            .then(() => {
                fixture = TestBed.createComponent(RegisterComponent);
                component = fixture.componentInstance;
            });
    }));

    it('should create the component', () => {
        expect(component).toBeTruthy();
    });

    it('should render the component correctly', () => {
        expect(fixture).toMatchSnapshot();
    });

    it('should have an invalid registration form when empty', () => {
        expect(component.registerForm.valid).toBeFalsy();
    });

    it('should have required validators on the form fields', () => {
        const firstNameField = component.registerForm.controls['firstName'];
        const lastNameField = component.registerForm.controls['lastName'];
        const emailField = component.registerForm.controls['email'];
        const password1Field = component.registerForm.controls['password1'];
        const firstNameErrors = firstNameField.errors || {};
        const lastNameErrors = lastNameField.errors || {};
        const emailErrors = emailField.errors || {};
        const passwordErrors = password1Field.errors || {};

        expect(firstNameErrors['required']).toBeTruthy();
        expect(lastNameErrors['required']).toBeTruthy();
        expect(emailErrors['required']).toBeTruthy();
        expect(passwordErrors['required']).toBeTruthy();
    });

    it('should have an invalid first name field when being empty', () => {
        const frstNameField = component.registerForm.controls['firstName'];

        expect(frstNameField.valid).toBeFalsy();
    });

    it('should have an invalid last name field when being empty', () => {
        const lastNameField = component.registerForm.controls['lastName'];

        expect(lastNameField.valid).toBeFalsy();
    });

    it('should have an invalid email field when being empty', () => {
        const emailField = component.registerForm.controls['email'];

        expect(emailField.valid).toBeFalsy();
    });

    it('should have an invalid password field when being empty', () => {
        const password1Field = component.registerForm.controls['password1'];

        expect(password1Field.valid).toBeFalsy();
    });

    it('should have an invalid status with two different passwords', () => {
        const password1Field = component.registerForm.controls['password1'];
        const password2Field = component.registerForm.controls['password2'];
        password1Field.setValue('test');
        password2Field.setValue('testt');
        const errors = password2Field.errors || {};

        expect(errors['validatePasswordConfirm']).toBeTruthy();
    });

    it('should emit the registratin credentials when submitting the registration form', () => {
        const email = 'test@test';
        const password = 'testtest';
        const firstName = 'test';
        const lastName = 'test;';

        expect(component.registerForm.valid).toBeFalsy();

        component.registerForm.controls['email'].setValue(email);
        component.registerForm.controls['password1'].setValue(password);
        component.registerForm.controls['password2'].setValue(password);
        component.registerForm.controls['firstName'].setValue(firstName);
        component.registerForm.controls['lastName'].setValue(lastName);
        component.registerForm.controls['dataProtection'].setValue(true);

        expect(component.registerForm.valid).toBeTruthy();

        let registerRequest: RegistrationCredentials;
        component.register.subscribe((value) => registerRequest = value);

        component.onRegister();

        expect(registerRequest!.email).toBe(email);
        expect(registerRequest!.password).toBe(password);
        expect(registerRequest!.firstName).toBe(firstName);
        expect(registerRequest!.lastName).toBe(lastName);
    });
});
