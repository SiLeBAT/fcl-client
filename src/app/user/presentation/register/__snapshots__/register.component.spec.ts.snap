// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterComponent should render the component correctly 1`] = `
<fcl-register
  changeRef={[Function ViewRef_]}
  pwStrength={[Function Number]}
  register={[Function EventEmitter_]}
  registerForm={[Function FormGroup]}
>
  <form
    novalidate=""
  >
    <div
      class="fcl-mat-card-content"
    >
      <div
        class="form-group fcl-form-firstname"
      >
        <mat-form-field
          appearance="standard"
          class="fcl-form-field mat-form-field mat-primary"
        >
          <div
            class="mat-form-field-wrapper"
          >
            <div
              class="mat-form-field-flex"
            >
              
              
              <div
                class="mat-form-field-infix"
              >
                <input
                  class="mat-input-element mat-form-field-autofill-control"
                  formcontrolname="firstName"
                  matinput=""
                  name="firstName"
                  required=""
                />
                <span
                  class="mat-form-field-label-wrapper"
                >
                  
                </span>
              </div>
              
            </div>
            
            <div
              class="mat-form-field-subscript-wrapper"
            >
              
              
            </div>
          </div>
        </mat-form-field>
      </div>
      <div
        class="form-group"
      >
        <mat-form-field
          appearance="standard"
          class="fcl-form-field mat-form-field mat-primary"
        >
          <div
            class="mat-form-field-wrapper"
          >
            <div
              class="mat-form-field-flex"
            >
              
              
              <div
                class="mat-form-field-infix"
              >
                <input
                  class="mat-input-element mat-form-field-autofill-control"
                  formcontrolname="lastName"
                  matinput=""
                  name="lastName"
                  required=""
                />
                <span
                  class="mat-form-field-label-wrapper"
                >
                  
                </span>
              </div>
              
            </div>
            
            <div
              class="mat-form-field-subscript-wrapper"
            >
              
              
            </div>
          </div>
        </mat-form-field>
      </div>
      <div
        class="form-group"
      >
        <mat-form-field
          appearance="standard"
          class="fcl-form-field mat-form-field mat-primary"
        >
          <div
            class="mat-form-field-wrapper"
          >
            <div
              class="mat-form-field-flex"
            >
              
              
              <div
                class="mat-form-field-infix"
              >
                <input
                  class="mat-input-element mat-form-field-autofill-control"
                  formcontrolname="email"
                  matinput=""
                  name="email"
                  required=""
                />
                <span
                  class="mat-form-field-label-wrapper"
                >
                  
                </span>
              </div>
              
            </div>
            
            <div
              class="mat-form-field-subscript-wrapper"
            >
              
              
            </div>
          </div>
        </mat-form-field>
      </div>
      <div
        class="form-group fcl-form-password"
      >
        <mat-form-field
          appearance="standard"
          class="fcl-form-field mat-form-field mat-primary"
        >
          <div
            class="mat-form-field-wrapper"
          >
            <div
              class="mat-form-field-flex"
            >
              
              
              <div
                class="mat-form-field-infix"
              >
                <input
                  class="mat-input-element mat-form-field-autofill-control"
                  formcontrolname="password1"
                  matinput=""
                  name="password1"
                  required=""
                  type="password"
                />
                <span
                  class="mat-form-field-label-wrapper"
                >
                  
                </span>
              </div>
              
            </div>
            
            <div
              class="mat-form-field-subscript-wrapper"
            >
              
              
            </div>
          </div>
        </mat-form-field>
      </div>
      <div
        class="form-group"
      >
        <password-strength-meter>
          <div>
            <div
              class="strength-meter"
            >
              <div
                class="strength-meter-fill"
              />
            </div>
            
          </div>
        </password-strength-meter>
        
      </div>
      <div
        class="form-group"
      >
        <mat-form-field
          appearance="standard"
          class="fcl-form-field mat-form-field mat-primary"
        >
          <div
            class="mat-form-field-wrapper"
          >
            <div
              class="mat-form-field-flex"
            >
              
              
              <div
                class="mat-form-field-infix"
              >
                <input
                  class="mat-input-element mat-form-field-autofill-control"
                  formcontrolname="password2"
                  matinput=""
                  name="password2"
                  required=""
                  type="password"
                />
                <span
                  class="mat-form-field-label-wrapper"
                >
                  
                </span>
              </div>
              
            </div>
            
            <div
              class="mat-form-field-subscript-wrapper"
            >
              
              
            </div>
          </div>
        </mat-form-field>
      </div>
      <div
        class="form-group fcl-form-checkbox"
      >
        <mat-checkbox
          class="mat-checkbox mat-accent"
          formcontrolname="dataProtection"
          required=""
        >
          <label
            class="mat-checkbox-layout"
          >
            <div
              class="mat-checkbox-inner-container"
            >
              <input
                class="mat-checkbox-input cdk-visually-hidden"
                type="checkbox"
              />
              <div
                class="mat-checkbox-ripple mat-focus-indicator mat-ripple"
                matripple=""
              >
                <div
                  class="mat-ripple-element mat-checkbox-persistent-ripple"
                />
              </div>
              <div
                class="mat-checkbox-frame"
              />
              <div
                class="mat-checkbox-background"
              >
                <svg
                  class="mat-checkbox-checkmark"
                  focusable="false"
                  version="1.1"
                  viewBox="0 0 24 24"
                  xml:space="preserve"
                >
                  <path
                    class="mat-checkbox-checkmark-path"
                    d="M4.1,12.7 9,17.6 20.3,6.3"
                    fill="none"
                    stroke="white"
                  />
                </svg>
                <div
                  class="mat-checkbox-mixedmark"
                />
              </div>
            </div>
            <span
              class="mat-checkbox-label"
            >
              <span
                style="display:none"
              >
                 
              </span>
               I have read and accept the 
              <a
                class="fcl-policy-link"
                target="_blank"
              >
                 Privacy Policy 
              </a>
               * 
            </span>
          </label>
        </mat-checkbox>
        
      </div>
      <div
        class="form-group fcl-form-checkbox"
      >
        <mat-checkbox
          class="fcl-newsletter-checkbox mat-checkbox mat-accent"
          formcontrolname="newsletter"
        >
          <label
            class="mat-checkbox-layout"
          >
            <div
              class="mat-checkbox-inner-container"
            >
              <input
                class="mat-checkbox-input cdk-visually-hidden"
                type="checkbox"
              />
              <div
                class="mat-checkbox-ripple mat-focus-indicator mat-ripple"
                matripple=""
              >
                <div
                  class="mat-ripple-element mat-checkbox-persistent-ripple"
                />
              </div>
              <div
                class="mat-checkbox-frame"
              />
              <div
                class="mat-checkbox-background"
              >
                <svg
                  class="mat-checkbox-checkmark"
                  focusable="false"
                  version="1.1"
                  viewBox="0 0 24 24"
                  xml:space="preserve"
                >
                  <path
                    class="mat-checkbox-checkmark-path"
                    d="M4.1,12.7 9,17.6 20.3,6.3"
                    fill="none"
                    stroke="white"
                  />
                </svg>
                <div
                  class="mat-checkbox-mixedmark"
                />
              </div>
            </div>
            <span
              class="mat-checkbox-label"
            >
              <span
                style="display:none"
              >
                 
              </span>
              <div
                class="fcl-text-wrap"
              >
                 I agree to receive occasional email newsletters on e.g. new features of FCL Web or user trainings. I can unsubscribe anytime by sending an email to 
                <a
                  class="fcl-policy-link"
                  href="mailto:foodrisklabs@bfr.bund.de"
                  target="_top"
                >
                   foodrisklabs@bfr.bund.de 
                </a>
                 . 
              </div>
            </span>
          </label>
        </mat-checkbox>
      </div>
      <div
        class="fcl-card-action fcl-card-action-container"
      >
        <div
          class="form-group fcl-submit-button"
        >
          <button
            class="mat-focus-indicator mat-raised-button mat-button-base"
            color="primary"
            mat-raised-button=""
            type="submit"
          >
            <span
              class="mat-button-wrapper"
            >
               Submit 
            </span>
            <span
              class="mat-button-ripple mat-ripple"
              matripple=""
            />
            <span
              class="mat-button-focus-overlay"
            />
          </button>
        </div>
        <div>
          <a
            class="mat-focus-indicator mat-button mat-button-base"
            mat-button=""
          >
            <span
              class="mat-button-wrapper"
            >
               Cancel 
            </span>
            <span
              class="mat-button-ripple mat-ripple"
              matripple=""
            />
            <span
              class="mat-button-focus-overlay"
            />
          </a>
        </div>
      </div>
    </div>
  </form>
</fcl-register>
`;
